packages/core-flows/src/order/workflows/create-orders.ts

```diff
@@ -0,0 +1,203 @@
+      variables: { variant_id: variantIds },
+    }).config({ name: "inventory-items" })
+
+    const confirmInventoryInput = transform(
+      { productVariantInventoryItems, salesChannelLocations, input, variants },
+      (data) => {
+        if (!data.input.items) {
+          return { items: [] }
+        }
+
+        if (!data.salesChannelLocations.length) {
+          throw new MedusaError(
+            MedusaError.Types.INVALID_DATA,
+            `Sales channel ${data.input.sales_channel_id} is not associated with any stock locations.`
+          )
+        }
+
+        const items = prepareConfirmInventoryInput({
+          product_variant_inventory_items: data.productVariantInventoryItems,
+          location_ids: data.salesChannelLocations[0].stock_locations.map(
+            (l) => l.id
+          ),
+          items: data.input.items!,
+          variants: data.variants.map((v) => ({
+            id: v.id,
+            manage_inventory: v.manage_inventory,
+          })),
+        })
+
+        return { items }
+      }
+    )
+
+    confirmInventoryStep(confirmInventoryInput)
+
+    // TODO: This is on par with the context used in v1.*, but we can be more flexible.
+    const pricingContext = transform(
+      { input, region, customerData },
+      (data) => {
+        return {
+          currency_code: data.input.currency_code ?? data.region.currency_code,
+          region_id: data.region.id,
+          customer_id: data.customerData.customer?.id,
+        }
+      }
+    )
+
+    const priceSets = getVariantPriceSetsStep({
+      variantIds,
+      context: pricingContext,
+    })
+
+    const orderInput = transform(
+      { input, region, customerData, salesChannel },
+      (data) => {
+        const data_ = {
+          ...data.input,
+          currency_code: data.input.currency_code ?? data.region.currency_code,
+          region_id: data.region.id,
+        }
+
+        if (data.customerData.customer?.id) {
+          data_.customer_id = data.customerData.customer.id
+          data_.email = data.input?.email ?? data.customerData.customer.email
+        }
+
+        if (data.salesChannel?.id) {
+          data_.sales_channel_id = data.salesChannel.id
+        }
+
+        return data_
+      }
+    )
+
+    const lineItems = transform({ priceSets, input, variants }, (data) => {
+      const items = (data.input.items ?? []).map((item) => {
+        const variant = data.variants.find((v) => v.id === item.variant_id)!
+
+        if (!variant) {
+          return prepareCustomLineItemData({
+            variant: {
+              ...item,
+            },
+            unitPrice: MathBN.max(0, item.unit_price),
+            quantity: item.quantity as number,
+            metadata: item?.metadata ?? {},
+          })
+        }
+
+        return prepareLineItemData({
+          variant: variant,
+          unitPrice: MathBN.max(
+            0,
+            item.unit_price ?? data.priceSets[item.variant_id!]?.amount
```

Comment:
Q: don't we want the min between those two?

Reply:
here if unit_price is provided it has to be used and replace the variant price, and the .max is to prevent negative values (this is being validated in v1)